inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/permute/permute_kernel_tile_8x8_4x4_fsv.cpp

```diff
@@ -0,0 +1,340 @@
+    const Datatype output_type = params.output.GetDType();
+
+    // i64 only supports tile size 4
+    if ((input_type == Datatype::INT64) || (output_type == Datatype::INT64)) {
+        return MIN_TILE_SIZE;
+    }
+
+    // supports 4x4 or 8x8 tiling
+    size_t rotating_dim = 0;
+    if (params.inputs[0].GetDims().size() == 4) {
+        rotating_dim = params.inputs[0].Y().v;
+    } else if (params.inputs[0].GetDims().size() == 5) {
+        rotating_dim = params.inputs[0].Z().v;
+    }
+
+    if (rotating_dim < DEFAULT_TILE_SIZE && params.inputs[0].Feature().v < DEFAULT_TILE_SIZE) {
+        return MIN_TILE_SIZE;
+    }
+
+    return DEFAULT_TILE_SIZE;
+}
+
+JitConstants PermuteKernel_tile_8x8_4x4_fsv::GetJitConstants(const permute_params& params, const CommonDispatchData& dispatchData) const {
+    auto jit = Parent::GetJitConstants(params, dispatchData);
+    const size_t f = params.inputs[0].Feature().v;
+    const size_t z = params.inputs[0].Z().v;
+    const size_t y = params.inputs[0].Y().v;
+    const size_t x = params.inputs[0].X().v;
+
+    const size_t tile_width = GetTileSize(params);
+    const size_t tile_height = tile_width;
+
+    // Note: this is default mode and different vector width is not being used now.
+    const uint64_t total_lws = dispatchData.lws[0] * dispatchData.lws[1] * dispatchData.lws[2];
+    const uint64_t transpose_buffer_size = tile_width*tile_height * total_lws;
+
+    const size_t input_ndims = params.inputs[0].GetDims().size();
+    const size_t output_ndims = params.output.GetDims().size();
+
+    const size_t fsv_alignment = GetFsvAlignment(params);
+
+    jit.AddConstant(MakeJitConstant("INPUT0_TILED_ORDER", GetTiledInputOrder(input_ndims)));
+    jit.AddConstant(MakeJitConstant("OUTPUT_TILED_ORDER", GetTiledOutputOrder(output_ndims)));
+
+    jit.AddConstant(MakeJitConstant("INPUT0_SIZE_FS", CEIL_DIV(f, fsv_alignment)));
+    jit.AddConstant(MakeJitConstant("INPUT0_SIZE_F", f));
+    jit.AddConstant(MakeJitConstant("TILE_WIDTH", tile_width));
+    jit.AddConstant(MakeJitConstant("TILE_HEIGHT", tile_height));
+    jit.AddConstant(MakeJitConstant("TILE_STRIDE", fsv_alignment * x));
+    jit.AddConstant(MakeJitConstant("FSV_ALIGNMENT", fsv_alignment));
+    jit.AddConstant(MakeJitConstant("GROUP_STRIDE", tile_width * x * fsv_alignment));
+    jit.AddConstant(MakeJitConstant("INPUTVTYPE", "CAT(INPUT0_TYPE, TILE_WIDTH)"));
+    jit.AddConstant(MakeJitConstant("OUTPUTVTYPE", "CAT(OUTPUT_TYPE, TILE_HEIGHT)"));
+    jit.AddConstant(MakeJitConstant("VLOAD", "CAT(vload, TILE_WIDTH)"));
+    jit.AddConstant(MakeJitConstant("VSTORE", "CAT(vstore, TILE_HEIGHT)"));
+    jit.AddConstant(MakeJitConstant("AS_INPUTVTYPE", "CAT(as_, INPUTVTYPE)"));
+    jit.AddConstant(MakeJitConstant("AS_OUTPUTVTYPE", "CAT(as_, OUTPUTVTYPE)"));
+    jit.AddConstant(MakeJitConstant("LOCAL_BUF_STRIDE", tile_width*tile_height));
+    jit.AddConstant(MakeJitConstant("TRANS_BUF_SIZE", transpose_buffer_size / tile_height));
```

Comment:
why calculating transpose_buffer_size as tile_height*tile_width * local ws and then divide again ? just tile_width * local_ws isnt enough?

Reply:
Fixed to use `tile_width*local_ws`